File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        bool flag = true;
        vector<TreeNode*> prev;
        prev.push_back(root);
        vector<vector<int>> res;
        if(root == NULL){
            return res;
        }
        while(flag){
            vector<TreeNode*> temp;
            vector<int> current;
            for(TreeNode* i: prev){
                current.push_back(i->val);
                if(i->left){
                    temp.push_back(i->left);
                }
                if(i->right){
                    temp.push_back(i->right);
                }
            }
            res.push_back(current);
            if(temp.size() == 0){
                flag = false;
            }
            else{
                prev = move(temp);
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/reverse-integer/submission-1.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        negetive = False
        if(x < 0):
            x = -x
            negetive = True
        
        while(x):
            temp = x%10
            x = x//10
            res = res*10 + temp
        if(negetive):
            res = -res
        if((-2**31) <= res <= (2**31 - 1)):
            return res
        else:
            return 0